File: app/src/main/java/ds/photosight/ui/widget/GalleryViewPager.java
package ds.photosight.ui.widget;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

public class GalleryViewPager extends ViewPager {

	public boolean zoomMode = false, needInit = true;

	private int x, y;
	private ClickListener mClickListener;


	public GalleryViewPager(Context context) {
		this(context, null);

	}


	public GalleryViewPager(final Context context, final AttributeSet attrs) {
		super(context, attrs);
	}


/*	public void setOnClickListener(ClickListener l) {
		mClickListener = l;
	}*/


	@Override
	public boolean onInterceptTouchEvent(MotionEvent e) {

		try {
			return super.onInterceptTouchEvent(e);
		} catch (IllegalArgumentException ex) {
			ex.printStackTrace();
			return false;
		}
	}



	public interface ClickListener {

		public void onClick();
	}

}

File: app/src/main/java/ds/photosight/ui/widget/GalleryViewPager.kt
package ds.photosight.ui.widget

import android.content.Context
import android.support.v4.view.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent

public class GalleryViewPager(context: Context, attrs: AttributeSet? = null) : ViewPager(context, attrs) {

/*    public var zoomMode: Boolean = false
    public var needInit: Boolean = true

    private val x: Int = 0
    private val y: Int = 0
    private val mClickListener: ClickListener? = null*/


    override fun onInterceptTouchEvent(e: MotionEvent): Boolean {
        try {
            return super.onInterceptTouchEvent(e)
        } catch (ex: IllegalArgumentException) {
            ex.printStackTrace()
            return false
        }
    }


    public trait ClickListener {
        public fun onClick()
    }

}